server/repository/src/db_diesel/item_row.rs

```diff
@@ -107,7 +110,15 @@ impl<'a> ItemRowRepository<'a> {
 
     pub fn find_one_by_id(&self, item_id: &str) -> Result<Option<ItemRow>, RepositoryError> {
         let result = item
-            .filter(id.eq(item_id))
+            .filter(id.eq(item_id).and(is_active.eq(true)))
```

Comment:
@roxy-dao @andreievg I'm working on `requisition_line` translator having the right item_id on push (#2636) and currently it's using this function. I was considering writing a new function below this `find_one_by_item_link_id` which for a given `item_link.id` returns the item related to `item_link.item_id`. I was wondering if it is generally better to not filter here though, depending on how `is_active` gets used in the future? On one hand it's used with item merging to preserve the original item records that get soft deleted in the merge. But I think it's conceivable that we might use it for soft deleting an item with merging anything in the future?

Anyway, now wondering why `find_one_by_id` applies this filter, but not `find_all` or `find_many_by_id`. Seems unclear why some would and some wouldn't filter for active items.

Reply:
Perhaps it should be up the gql query to specify the filter, rather than backend trying to predict what consumers of the API want?